### PR TITLE
[Feature][Depends on #339] Added an app tour. Added a joyRide tour to the application to help users to understand the dashboard features.

### PR DESCRIPTION
**Pull request by anvithks**
## **This PR depends on changes made in ~#329~ #339 . Please merge in order.**

**What type of PR is this?**
/kind new feature

**What this PR does / why we need it**:
This PR adds an application tour that tells users what the different sections of the Dashboard. This will help users to get a quick idea of what each link in the dashboard does without having to read a lot of documentation.

**Which issue(s) this PR fixes**:
This PR adds some of the functionality of  #327. A complete user guide and documentation will be added shortly.

**Test Report Added?**:
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
/kind TESTED

**Test Report**:
The demo video can be viewed here: [DEMO VIDEO](https://drive.google.com/file/d/1FTDo-Jc9O0uzmobgRjyH5NUcU9OHlFea/view)
tour-layout-1
tour-layout-2
tour-layout-3
tour-layout-4
tour-layout-5
tour-layout-6
tour-layout-7

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by kumarashit** [comment]
@anvith #329 is for TCs..why is this dependent on that..I didn't get the point
Also, i hope the help defs will be improved going ahead as some of the definitions may not be very intuitive

**Comment by anvithks** [comment]
@kumarashit That was a typo. The PR is dependent on #339 The PR title has the correct reference.

**Comment by anvithks** [comment]
> @anvith #329 is for TCs..why is this dependent on that..I didn't get the point
> Also, i hope the help defs will be improved going ahead as some of the definitions may not be very intuitive

The definitions for each step can be updated. You can comment with the suggested changes and I will update them.

File: src/app/app.component.html [hunk 1]
```diff
@@ -24,21 +24,24 @@
...
+                </div>
+                <div class="ui-g-1">
+                    <button type="button" pButton class="start-tour-button ui-button" (click)="startTour()" label="Start Tour"></button>
                 </div>
                 <div class="ui-g-2">
                     <div class="user-info">
                         <div>
                             <h1>Welcome {{ username }}</h1>
-                            <p-dropmenu  icon="fa-chevron-circle-down" [model]="dropMenuItems" styleClass="ui-dropmenu-large"></p-dropmenu>
+                            <p-dropmenu joyrideStep="homeUserProfile" stepPosition="bottom" [title]="'User menu'" [text]="'You can switch region, navigate to Manage AK/SK, Monitor Configuration and Logout'" icon="fa-chevron-circle-down" [model]="dropMenuItems" styleClass="ui-dropmenu-large"></p-dropmenu>
```
Comment (kumarashit) [inline src/app/app.component.html:44]:
```suggestion
                            <p-dropmenu joyrideStep="homeUserProfile" stepPosition="bottom" [title]="'User menu'" [text]="'You can switch region, navigate to Key Management System (AK/SK), Monitor Configuration and Logout'" icon="fa-chevron-circle-down" [model]="dropMenuItems" styleClass="ui-dropmenu-large"></p-dropmenu>
```

File: src/app/app.component.html [hunk 2]
```diff
@@ -68,6 +71,13 @@ <h1>{{menuItem.title}}</h1>
         </div>
     </div>
 </div>
+<ng-template #homeWelcomeContent>
+    <div class="tour-content">
+        SODA Dashboard provides a unified interface to the underlying capabilities from the SODA Framework components. This dashboard is built mainly to illustrate/showcase various use cases that could run on SODA ecosystem projects.
+
+        Technically, it is a direct client to SODA API and other projects which are getting integrated to SODA Ecosystem.
```
Comment (kumarashit) [inline src/app/app.component.html:78]:
```suggestion
        It is a direct client to SODA API and other projects which are getting integrated to SODA Ecosystem.
```

File: src/app/app.component.ts [hunk 3]
```diff
@@ -70,83 +72,152 @@ export class AppComponent implements OnInit, AfterViewInit {
...
+    tourSteps = [];
 
     menuItems_tenant = [
         {
             "title": "Home",
             "description": "Resource statistics",
-            "routerLink": "/home"
+            "routerLink": "/home",
+            "joyrideStep" : "menuHome",
+            "text" : "This page shows you the overall view of the backends available, volumes provisioned, buckets created and migrations performed."
```
Comment (kumarashit) [inline src/app/app.component.ts:83]:
```suggestion
            "text" : "This page shows you the overall view of the backends available, volumes provisioned, buckets created and object storage migrations performed."
```

File: src/app/app.component.ts [hunk 4]
```diff
@@ -70,83 +72,152 @@ export class AppComponent implements OnInit, AfterViewInit {
...
         },
         {
             "title": "Profile",
             "description": "Profiles",
-            "routerLink": "/profile"
+            "routerLink": "/profile",
+            "joyrideStep" : "menuProfile",
+            "text" : "A profile is a set of configurations on service capabilities (including resource tuning, QoS) of storage resources. A profile must be specified when volume is created."
```
Comment (kumarashit) [inline src/app/app.component.ts:90]:
```suggestion
            "text" : "A profile is a set of configurations on service capabilities (including resource tuning, QoS) of storage resources. A profile must be specified when volume or fileshare is created."
```

File: src/app/app.component.ts [hunk 5]
```diff
@@ -70,83 +72,152 @@ export class AppComponent implements OnInit, AfterViewInit {
...
     ]
 
     menuItems_admin = [
         {
             "title": "Home",
             "description": "Resource statistics",
-            "routerLink": "/home"
+            "routerLink": "/home",
+            "joyrideStep" : "menuHome",
+            "text" : "This page shows you the overall view of the backends available, volumes provisioned, buckets created and migrations performed."
```
Comment (kumarashit) [inline src/app/app.component.ts:128]:
```suggestion
            "text" : "This page shows you the overall view of the backends available, volumes provisioned, buckets created and object storage migrations performed."
```

File: src/app/business/home/home.component.html [hunk 6]
```diff
@@ -63,8 +63,8 @@
             </svg>
         </div>
     </div>
-    <div style="display:flex;">
-        <div style="flex:1;padding:20px;">
+    <div style="display:flex;" >
+        <div joyrideStep="homeResourceCard" stepPosition="top" text="You can see the number of volumes provisioned, buckets created and Filesystems."  style="flex:1;padding:20px;">
```
Comment (kumarashit) [inline src/app/business/home/home.component.html:67]:
```suggestion
        <div joyrideStep="homeResourceCard" stepPosition="top" text="You can see the number of volumes provisioned, buckets created and Fileshares created."  style="flex:1;padding:20px;">
```

**Pull request merged**